python/thunder/rdds/series.py

```diff
@@ -582,18 +582,80 @@ def query(self, inds, var='inds', order='F', isOneBased=True):
 
         return keys, values
 
-    def meanOfRegion(self, keys):
-        """Takes the mean of Series values within a single region specified by the passed keys.
+    def __maskToKeys(self, mask, returnNested=False):
+        """Helper method to validate and convert a binary mask to a set of keys for use in
+        mean of/by region(s).
+
+        If returnNested is true, will return a sequence of sequences of keys, suitable for use
+        in meanByRegions. If returnNested is true and an integer or uint mask is passed, a separate set
+        of keys will be returned for each unique nonzero value in the mask, sorted in numeric order of the
+        mask values.
+
+        If returnNested is false, a single sequence of keys will be returned. The keys will be the indices
+        of all nonzero values of the passed mask.
+
+        Parameters
+        ----------
+        mask: ndarray
+
+        returnNested: boolean, optional, default False
+
+        Returns
+        -------
+        sequence of subscripted indices if returnNested is false,
+        sequence of sequence of subscripted indices if returnNested is true
+        """
+        # argument type checking
+        if not isinstance(mask, ndarray):
+            raise ValueError("Mask should be numpy ndarray, got: '%s'" % str(type(mask)))
+        # check for matching shapes only if we already know our own shape; don't trigger action otherwise
+        if self._dims:
```

Comment:
This is a bit weird. I see the rationale for not triggering. However, if the user violates this assumption, and we don't check, it may actually run to completion with no errors (for example, it may just not find the full set of regions the user expected), right? Normally, I'm fine avoiding an expensive pre-computation if we know the user will get an error on the workers down the road. But here that might not be the case. So maybe we always want to check ahead of time? Unfortunately, there's no way to also check that the keys are contiguous, which is another implicit assumption of using the mask...